src/app/modules/catalog/components/tab-content/tab-content.component.ts

```diff
@@ -0,0 +1,63 @@
+import {
+    AfterViewInit,
+    ChangeDetectionStrategy,
+    Component,
+    Input,
+    OnInit,
+    Renderer2,
+    ViewChild,
+} from '@angular/core';
+import * as Models from '../../state/catalog.models';
+
+@Component({
+    selector: 'app-tab-content',
+    templateUrl: './tab-content.component.html',
+    styleUrls: ['./tab-content.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TabContentComponent implements OnInit, AfterViewInit {
+    @Input() offersData: Models.OffersCategories;
+    @Input() offersPromos: Array<Models.PromosData>;
+    @ViewChild('overlay') overlay: { nativeElement: string };
+
+    categories: Array<Models.CategoryName>;
+    data: Array<Models.CategoriesData>;
+    filteredData: Array<Models.CategoriesData>;
+
+    private element: { nativeElement: string };
+
+    constructor(private readonly renderer: Renderer2) {}
+
+    ngOnInit(): void {
+        this.categories = this.offersData.categoryNames;
+        this.data = this.offersData.data;
+    }
+
+    ngAfterViewInit(): void {
+        this.element = this.overlay;
+    }
+
+    addClass(className: string, element: { nativeElement: string }) {
```

Comment:
this functionality shouldn't be here, tab content - just content and nothing more
so, pls rework